### DISCUSSION TITLE
Base page template

### DISCUSSION THREAD

**Discussion by boxadesign** [ONSdigital/design-system#195] [Feedback]
Use this place to discuss the [base page template](https://ons-design-system.netlify.com/foundations/page-template/) in the [ONS Design System](https://ons-design-system.netlify.com)

**Comment by jrbarnes9** [comment]
The page template example with save and complete - button doesn't display in the footer < 740

**Comment by jrbarnes9** [comment]
No config for the masthead logo href in the page template

**Comment by jrbarnes9** [comment]
Width and height of svgs lost when rendered in the browser
<img width="773" alt="Screenshot 2020-02-11 at 14 13 54" src="https://user-images.githubusercontent.com/43346934/74244126-dea19600-4cd8-11ea-8306-6f6b8be50eb3.png">

**Comment by jrbarnes9** [comment]
` prePage` block should be `preMain` in exploded view of page template

![Screenshot 2020-06-19 at 11 27 36](https://user-images.githubusercontent.com/43346934/85123304-eee5f680-b21f-11ea-82a1-5f029da22808.png)

**Comment by jrbarnes9** [comment]
Unlinked references in page template variable tables

**Comment by jrbarnes9** [comment]
Increase `page__main` margin-top to 2.5rem (over 740) and 1.5rem (under 740).

Spacing is currently too tight.

<img width="886" alt="Screenshot 2020-09-03 at 08 49 09" src="https://user-images.githubusercontent.com/43346934/92086627-5bee3e80-edc2-11ea-9e2c-f30de97f920a.png">

**Comment by jrbarnes9** [comment]
Save and sign out button in footer of page template example ('Customised page template') is missing exit icon

<img width="279" alt="Screenshot 2020-09-17 at 09 18 20" src="https://user-images.githubusercontent.com/43346934/93439794-e062b600-f8c6-11ea-9204-f77be5afd905.png">

**Comment by Paul-Joel** [comment]
Is there a way to provide a 'nonce' for the inline script calls without using Talisman? The existing if statements in the layout/_template.njk only allow for the talisman call to add a nonce as far as I can tell

**Comment by jrbarnes9** [comment]
Update page template languages logic.  If more than one other language, hide language toggles in masthead, and show in footer (at appropriate breakpoint)

**Comment by jrbarnes9** [comment]
Broken footer in census page template

https://ons-design-system.netlify.app/styles/page-template/examples/census/

![Screenshot 2020-12-01 at 16 59 52](https://user-images.githubusercontent.com/43346934/100771838-caa95b00-33f6-11eb-81a0-900d745e7634.png)

**Reply by jrbarnes9** [reply]
Resolved with #2299

**Comment by jrbarnes9** [comment]
Set page contents to be displayed right-to-left using css `direction: rtl`

**Reply by jrbarnes9** [reply]
#1296 resolved

**Comment by jrbarnes9** [comment]
Reduce `margin-top` when `page__main` contains a child with `panel--error` 

Visual vertical spacing improvement for when an error summary panel is displayed. Equal spacing between header to breadcrumbs, and breadcrumbs to error panel.

From...
![Screenshot 2021-01-19 at 08 21 20](https://user-images.githubusercontent.com/43346934/105008692-76ebe300-5a31-11eb-8ef7-e933f41acb6a.png)

To...
![Screenshot 2021-01-19 at 08 21 40](https://user-images.githubusercontent.com/43346934/105008718-7fdcb480-5a31-11eb-9800-26e69482cb81.png)

**Reply by jrbarnes9** [reply]
Raised #1302

**Comment by jrbarnes9** [comment]
In page template, the documented `block: page` should be `block: pageContent`

**Reply by jrbarnes9** [reply]
#1491 resolved

**Comment by trevorsaint** [comment]
We can’t have a full width component that sits outside of the grid due to the way the page template is configured.

I was able to override the default grid output by adding the following to the pageConfig

```"pageColNumber": '12'```

**What we need**

<img width="1514" alt="extended" src="https://user-images.githubusercontent.com/4989027/122934718-b1063b80-d367-11eb-9bfc-be3f7d711054.png">

**What we get**

<img width="1464" alt="locked-inside-grid" src="https://user-images.githubusercontent.com/4989027/122934744-b794b300-d367-11eb-9adc-f36359817e63.png">

We could remove the pageConfig completely, but then we lose all of the other good things that come with it.

Ideally we want a way to remove grid usage by default, and then control items manually inside

```{% block main %}```

**Reply by jrbarnes9** [reply]
#1572 raised

**Comment by jrbarnes9** [comment]
Page template example shows `scripts` block but not documented in table. 

Also a typo in table on `bodyEnd` row: 
```
Add content just after just before the closing </body> element
```

**Reply by jrbarnes9** [reply]
#1773 raised

**Comment by JoelClemence** [comment]
Just tried to implement a new page and the example on the site says the template has path: `foundations/layout/template/_template.njk` does not exist guessing that `foundations/layout` have effectively been removed, I got it working with `page-templates/_template.njk` so the documentation probably needs to be updated

**Reply by jrbarnes9** [reply]
Raised #1789

**Reply by boxadesign** [reply]
https://github.com/ONSdigital/design-system/pull/1792

**Comment by jrbarnes9** [comment]
Header `variants` parameter not being passed to base page template

**Reply by jrbarnes9** [reply]
Raised #2272

**Comment by jrbarnes9** [comment]
Base page template custom example not using mobile serviceLinks menu for languages

**Reply by jrbarnes9** [reply]
Raised #2410

**Comment by jrbarnes9** [comment]
'Save and sign out' button missing from footer in custom base page template example

**Reply by jrbarnes9** [reply]
Raised #2539

**Comment by jrbarnes9** [comment]
Page template blocks exploded view example shows breadcrumbs within `preMain`, but they are placed before that when using page config

**Reply by jrbarnes9** [reply]
Raised #2542

**Comment by AlanHONS** [comment]
Found issue in https://github.com.mcas.ms/ONSdigital/prototype-kit-template where within https://github.com.mcas.ms/ONSdigital/prototype-kit-template/blob/main/src/prototypes/example/views/_main.njk 
Line 26 "OGLLink" has not been updated to latest all lowercase "oglLink"
Which means all footers in Prototype Kit pages are visually missing expected "All content is available under the Open Government Licence v3.0, except where otherwise stated" footer content

**Reply by rmccar** [reply]
Hi Alan, I have incorporated this fix into the upgrade to v73 of the DS. It will be fixed in the next release of the prototype kit template

**Comment by AlanHONS** [comment]
Small error in Design System page, that [Customising with variables](https://service-manual.ons.gov.uk/design-system/foundations/base-page-template#customising-with-variables) sample doesn't actually include stated use of 'related content' component atm/anymore. 
All other mentioned used components are however correct and present.

<img width="949" height="1146" alt="image" src="https://github.com/user-attachments/assets/81d2a13e-4781-4c06-a2dd-6e5260b2b533" />

**Reply by MagdalenaLarge** [reply]
@AlanHONS Hello,
thank you for highlighting this to us. We will update content and remove the reference to related content.
